### ISSUE TITLE
`GOOS=js GOARCH=wasm go test ./...` fails

### ISSUE THREAD

**Issue by hajimehoshi** [hajimehoshi/ebiten#2972]
### Ebitengine Version

v2.7.2

### Operating System

- [ ] Windows
- [X] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android
- [ ] iOS
- [ ] Nintendo Switch
- [ ] PlayStation 5
- [ ] Xbox
- [X] Web Browsers

### Go Version (`go version`)

go version go1.22.2 darwin/arm64

### What steps will reproduce the problem?

1. Set up the wasmbrowsertest:

```sh
go install github.com/agnivade/wasmbrowsertest@ee76d31b7b9b1645576c1f51fec4c09fe6cf1bb3`
cd $(go env GOPATH)/bin
ln -s wasmbrowsertest go_js_wasm_exec
```

2. Run the test for browsers

```sh
GOOS=js GOARCH=wasm go test .
```

### What is the expected result?

The tests pass

### What happens instead?

The tests fail:

```
panic: opengl: getContext for webgl2 failed

goroutine 1 [running]:
github.com/hajimehoshi/ebiten/v2/internal/testing.MainWithRunLoop(0x185bd60)
        /Users/hajimehoshi/ebiten/internal/testing/testing.go:48 +0x10
github.com/hajimehoshi/ebiten/v2_test.TestMain(...)
        /Users/hajimehoshi/ebiten/image_test.go:58
main.main()
        _testmain.go:373 +0xc
exit with status 2
FAIL    github.com/hajimehoshi/ebiten/v2        0.681s
FAIL
```

### Anything else you feel useful to add?

This is odd since we have not seen this issue until today. This was found in a PR https://github.com/hajimehoshi/ebiten/pull/2971

**Comment by hajimehoshi** [comment]
`WASM_HEADLESS=off` seems resolve the issue (but why?)

**Comment by hajimehoshi** [comment]
Also I found wasm tests didn't work on Windows due to too long environment variables. Let's fix this at the same time.

**Comment by hajimehoshi** [comment]
Fixed by f43c237c00e36e7a0c37e3c3713334129d6ac857
